python/lsst/obs/pfs/ingest.py

```diff
@@ -550,6 +550,62 @@ class PfsIngestTask(IngestTask):
     ArgumentParser = PfsIngestArgumentParser
     _DefaultName = "ingestPfs"
 
+    def findPfsConfig(self, dirName, fileInfo):
+        """Find a PfsConfig file
+
+        We search for the PfsConfig file in the following places, in order:
+        * ``dirName/``
+        * ``dirName/<dateObs>/pfsConfig/``
+
+        If the PfsConfig file is not found there, we search for a PfsDesign
+        file in the ``dirName`` directory, and use that to write a PfsConfig
+        file in the ``dirName/<dateObs>/pfsConfig`` directory.
+
+        Parameters
+        ----------
+        dirName : `str`
+            (Base) directory in which the file resides.
+        fileInfo : `dict`
+            Key-value pairs defining the file.
+
+        Returns
+        -------
+        path : `str`
+            Path to the PfsConfig file.
+
+        Raises
+        ------
+        RuntimeError
+            If no PfsConfig or PfsDesign file can be found.
+        """
+        pfsDesignId = fileInfo["pfsDesignId"]
+        visit = fileInfo["visit"]
+        fileName = PfsConfig.fileNameFormat % (pfsDesignId, visit)
+
+        path = os.path.join(dirName, fileName)
+        if os.path.exists(path):
+            return path
+        dateDirName = os.path.join(dirName, fileInfo["dateObs"], "pfsConfig")
+        path = os.path.join(dateDirName, fileName)
+        if os.path.exists(path):
+            return path
+
+        self.log.debug(
+            "Unable to find PfsConfig file for pfsDesignId=0x%016x, visit=%d; searching for pfsDesign file",
+            pfsDesignId,
+            visit,
+        )
+
+        designName = os.path.join(dirName, PfsDesign.fileNameFormat % (pfsDesignId,))
+        if not os.path.exists(designName):
+            raise RuntimeError(f"Unable to find PfsConfig or PfsDesign for pfsDesignId=0x{pfsDesignId:016x}")
+        design = PfsDesign.read(pfsDesignId, dirName)
+        if not os.path.exists(dateDirName):
+            os.makedirs(dateDirName)
+        PfsConfig.fromPfsDesign(design, visit, design.pfiNominal).write(dateDirName)
+        assert os.path.exists(path), f"Failed to write pfsConfig at expected path: {path}"
```

Comment:
Would be good to add a ``log.warn`` message to say that a new ``PfsConfig`` instance will be created, and its path, if one doesn't already exist. I'm just concerned when a ``PfsConfig`` file is created implicitly and unexpectedly.